### DISCUSSION TITLE
whatsNew goes missing after successful upload

### DISCUSSION THREAD

**Discussion by tmxkevingriesbach** [fastlane/fastlane#16862] [Q&A]
Hello,

I am currently stuck on a problem that makes me want to pull my hair out and I can't figure out what is happening. I am trying to upload testflights for two Apps.

1. The first one uploads just fine and the changelog text in whatsNew gets released to testers afterwards.
1. The second one uploads just fine too and the changelog/whatsNew text is nowhere to be seen on App Store Connect, but Output suggests it got set.

If there's anybody able to point me in any direction, I would be immensely grateful.

Part of my Fastfile (same for App 1 and 2):
```
require 'i18n'
I18n.config.available_locales = :en
changelog = File.read(varPathToChangelog)
changelog = I18n.transliterate(changelog)
[...]
puts "--- CHANGELOG ---"
puts changelog
puts "--- CHANGELOG ---"
upload_to_testflight(changelog: changelog)
```

The Output for App 2:
```
[13:23:13]: --- CHANGELOG ---
[13:23:13]: XYZ ABC DEF
ABC DEF GHI JKL MNO PQR
[13:23:13]: --- CHANGELOG ---
[...]
[13:42:21]: Waiting for App Store Connect to finish processing the new build (1.0.0 - 123456) for IOS
[13:42:51]: [32mSuccessfully finished processing the build 1.0.0 - 123456 for IOS[0m
[13:42:52]: [32mSuccessfully set the changelog for build[0m
[13:42:52]: Distributing new build to testers: 1.0.0 - 123456
[13:42:52]: [32mSuccessfully distributed build to Internal testers 🚀[0m
```

I am on fastlane 2.152.0 with no plugins, if that makes any difference.

Thank you very much!

**Comment by joshdholtz** [comment]
@tmxkevingriesbach This is super strange! I don’t know if I’ve seen this before. The initial hunch is that maybe something with the language that its being uploaded as? Do both apps have the same default language? Is the changelog maybe appearing under a different language? 🤔

**Reply by tmxkevingriesbach** [reply]
Unfortunately not. They are both german and also their only languages.
Is there some extra logging I could enable or something like that?

**Reply by tmxkevingriesbach** [reply]
Enabling verbose-logging did not reveal any new information. If you have any other ideas, please let me know. This is starting to drive me insane 🤯

**Reply by joshdholtz** [reply]
@tmxkevingriesbach Hmmmmm.... I’m going to change one of my test apps to only German to see if I can replicate this! My hunch is there is something weird going on there. 

Oh, another question! Does this change log to happen to have any special characters or anything in it? Does it work if you change the changelog to the exact changelog that the other app uses?

**Reply by tmxkevingriesbach** [reply]
@joshdholtz I did a long test-run with different texts, but ended with just using "Bugfixes". This worked for my first app, but not the second one.
I also just checked both logs again and noticed only one real difference. The time to process the second app is a third of the first one. It is quite a bit smaller and simpler, so I'd understand why, but could that change something maybe?

**Comment by tmxkevingriesbach** [comment]
@joshdholtz I'd guess you did not, by chance, find anything since then? We are still having this issue with this app unfortunately :(

**Comment by ingra62** [comment]
I have the exact same problem. Have tried a variety of things [walgreenslistens](https://www.walgreenslistens.biz/).
